tests/test_simulation.py

```diff
@@ -1032,9 +1036,13 @@ def test_mask_equals_remove():
             ValueError,
             match="The ids of qubits targeted in SLM mask",
         ):
-            QutipEmulator(sampler.sample(seq_masked), reg_two, MockDevice)
+            QutipEmulator(
+                sampler.sample(seq_masked),
+                reg_two,
+                replace(MockDevice, dmm_objects=(DMM(),)),
+            )
```

Comment:
Some of these `replace`s are conflicting with changes in my PR. Since we are going to have to delete them anyway, may I suggest we do that now and instead add this at the begining:
```
MockDevice = replace(MockDevice, dmm_objects=(DMM(),))
```

This way we only have to delete one line